OpenRA.Game/FileFormats/IdxReader.cs

```diff
@@ -0,0 +1,44 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.IO;
+using OpenRA.FileSystem;
+
+namespace OpenRA.FileFormats
+{
+	public class IdxReader
+	{
+		public readonly int SoundCount;
+		public List<IdxEntry> Entries;
+
+		public IdxReader(Stream s)
+		{
+			s.Seek(0, SeekOrigin.Begin);
+
+			var id = s.ReadASCII(4);
+
+			if (id != "GABA")
+				throw new InvalidDataException(string.Format("Unable to load Idx file, did not find magic id, found {0} instead", id));
```

Comment:
Please use our `.F()` extension method instead of `string.Format` for consistency with the rest of the codebase.